tests/components/nice_go/test_config_flow.py

```diff
@@ -109,3 +111,71 @@ async def test_duplicate_device(
     )
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+async def test_reauth(
```

Comment:
It would be nice to see a coordinator test as well that the `reauth` flow is started if the resulting exception in the coordinator is `ConfigEntryAuthFailed`. I think you have "similar" tests already in `test_init.py`